cli/internal/cmd/iamcreateaws.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/edgelesssys/constellation/v2/cli/internal/cloudcmd"
+	"github.com/edgelesssys/constellation/v2/internal/cloud/cloudprovider"
+	"github.com/spf13/cobra"
+)
+
+// newIAMCreateAWSCmd returns a new cobra.Command for the iam create aws command.
+func newIAMCreateAWSCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "aws",
+		Short: "Create IAM configuration on AWS for your Constellation cluster",
+		Long:  "Create IAM configuration on AWS for your Constellation cluster.",
+		Args:  cobra.ExactArgs(0),
+		RunE:  runIAMCreateAWS,
+	}
+
+	cmd.Flags().String("prefix", "", "Name prefix for all resources.")
+	must(cobra.MarkFlagRequired(cmd.Flags(), "prefix"))
+	cmd.Flags().String("zone", "", "AWS availability zone the resources will be created in (e.g. us-east-2a). Find available zones here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-availability-zones. Note that we do not support every zone / region. You can find a list of all supported regions in our docs.")
+	must(cobra.MarkFlagRequired(cmd.Flags(), "zone"))
+	cmd.Flags().Bool("yes", false, "Create the IAM configuration without further confirmation")
+
+	return cmd
+}
+
+func runIAMCreateAWS(cmd *cobra.Command, args []string) error {
+	spinner := newSpinner(cmd.ErrOrStderr())
+	defer spinner.Stop()
+	creator := cloudcmd.NewIAMCreator(spinner)
+
+	return iamCreateAWS(cmd, spinner, creator)
+}
+
+func iamCreateAWS(cmd *cobra.Command, spinner spinnerInterf, creator iamCreator) error {
+	// Get input variables.
+	awsFlags, err := parseAWSFlags(cmd)
+	if err != nil {
+		return err
+	}
+
+	// Confirmation.
+	if !awsFlags.yesFlag {
+		cmd.Printf("The following IAM configuration will be created:\n")
+		cmd.Printf("Region:\t%s\n", awsFlags.region)
+		cmd.Printf("Name Prefix:\t%s\n", awsFlags.prefix)
+		ok, err := askToConfirm(cmd, "Do you want to create the configuration?")
+		if err != nil {
+			return err
+		}
+		if !ok {
+			cmd.Println("The creation of the configuration was aborted.")
+			return nil
+		}
+	}
+
+	// Creation.
+	spinner.Start("Creating", false)
+	iamFile, err := creator.Create(cmd.Context(), cloudprovider.AWS, &cloudcmd.IAMConfig{
+		AWS: cloudcmd.AWSIAMConfig{
+			Region: awsFlags.region,
+			Prefix: awsFlags.prefix,
+		},
+	})
+	spinner.Stop()
+	if err != nil {
+		return err
+	}
+
+	cmd.Printf("region:\t%s\n", awsFlags.region)
+	cmd.Printf("zone:\t%s\n", awsFlags.zone)
+	cmd.Printf("iamProfileControlPlane:\t%s\n", iamFile.AWSOutput.ControlPlaneInstanceProfile)
+	cmd.Printf("iamProfileWorkerNodes:\t%s\n", iamFile.AWSOutput.WorkerNodeInstanceProfile)
+	cmd.Println("Your IAM configuration was created successfully. Please fill the above values into your configuration file.")
```

Comment:
As a potential follow up task: Have we thought about filling in the values directly into the configuration file? I imagine most users want to do this anyway, and `iam create` could get a flag `--no-fill-in` to prevent this in case this is not desired.

Reply:
I talked to Leo about this and the current decision was that anyone should be able to setup the IAM on an independent machine. However, I was also a supporter of directly filling the values into the config or at least offer a flag to enable automatic filling. For now, I think we should add the IAM command as is and create another ticket on the board.

Reply:
Tracked in AB#2706 now

Reply:
That makes sense!